go/arrow/csv/reader.go

```diff
@@ -76,6 +86,49 @@ func NewReader(r io.Reader, schema *arrow.Schema, opts ...Option) *Reader {
 	return rr
 }
 
+func (r *Reader) readHeader() error {
+	records, err := r.r.Read()
+	if err != nil {
+		return errors.New("failed to read file")
```

Comment:
I would create a global error value with a nice name and a nicer error message, specifying that we couldn't read the header.